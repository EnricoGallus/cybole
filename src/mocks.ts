export function registerElectron() {
    const electronMock = {
        openDialog: jest.fn().mockReturnValue(Promise.resolve({ filePaths: ['directory'] })),
        getFiles: jest.fn(),
        readFile: jest.fn(),
        writeFile: jest.fn(),
        getAppDescription: jest.fn().mockReturnValue(Promise.resolve('Cybole v0.2')),
        openContextMenu: jest.fn(),
        on: jest.fn(),
    };

    window.electron = electronMock;
    return electronMock;
}
