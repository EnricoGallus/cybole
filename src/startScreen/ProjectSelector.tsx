import React, {ReactNode} from 'react';
import {useNavigate} from 'react-router-dom';
import {Menu} from "primereact/menu";
import {MenuItemCommandParams} from "primereact/menuitem";

type PropsType = {
    children: ReactNode;
};

const ProjectSelector = (props: PropsType) => {
    const navigation = useNavigate();
    const {children} = props;
    const handleMenu = (route: string) => {
        navigation(route);
    };

    const menuItems = [
        {
            label: 'Menu',
            items: [
                {
                    label: 'Projects',
                    icon: 'pi pi-align-justify',
                    command: () => {
                        handleMenu('/')
                    }
                },
                {
                    label: 'Settings',
                    icon: 'pi pi-cog',
                    command: () => {
                        handleMenu('/settings')
                    }
                }
            ]
        }
    ];

    return (
        <div className="container">
            <div className="grid">
                <div className="card col-12">
                    <h5>Projects</h5>
                </div>
            </div>
            <div className="grid">
                <div className="col-3">
                    <Menu model={menuItems}/>
                </div>
                <div className="col-9">
                    {children}
                </div>
            </div>
        </div>
    );
};

export default ProjectSelector;
