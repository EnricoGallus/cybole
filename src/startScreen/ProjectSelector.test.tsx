import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ProjectSelector from './ProjectSelector';
import { registerElectron } from '../tests/mocks';

describe('ProjectSelector', () => {
    test('renders component', () => {
        const electronMock = registerElectron();
        const { container } = render(<ProjectSelector children={<div data-testid="project-open" />} />, {
            wrapper: MemoryRouter,
        });

        expect(container).toMatchSnapshot();
        expect(electronMock.getAppDescription).toBeCalled();
    });
});
