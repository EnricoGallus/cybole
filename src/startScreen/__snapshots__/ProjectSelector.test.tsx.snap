// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSelector renders component 1`] = `
<div>
  <section
    class="jsx-42076441 "
  >
    <header
      class="jsx-3963558373 "
    >
      <h2>
        Projects
      </h2>
    </header>
    <main
      class="jsx-1899865532 "
    >
      <div
        class="jsx-3633143255 item jsx-478091869"
      >
        <div
          class="jsx-2426585010 item xs jsx-2684866274"
        >
          <div
            class="jsx-1887233099 btn-group vertical"
          >
            <button
              class="jsx-1249061632 btn"
              disabled=""
              type="button"
            >
              <div
                class="text"
              />
            </button>
            <button
              class="jsx-489100405 btn"
              type="button"
            >
              <span
                class="jsx-643337184 icon"
              >
                <svg
                  fill="none"
                  height="24"
                  shape-rendering="geometricPrecision"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  style="color: currentColor;"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M8 6h13M8 12h13M8 18h13M3 6h.01M3 12h.01M3 18h.01"
                  />
                </svg>
              </span>
              <div
                class="jsx-887841902 text left"
              >
                Projects
              </div>
            </button>
            <button
              class="jsx-489100405 btn"
              type="button"
            >
              <span
                class="jsx-643337184 icon"
              >
                <svg
                  fill="none"
                  height="24"
                  shape-rendering="geometricPrecision"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  style="color: currentColor;"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                  <path
                    d="M19.4 15a1.65 1.65 0 00.33 1.82l.06.06a2 2 0 010 2.83 2 2 0 01-2.83 0l-.06-.06a1.65 1.65 0 00-1.82-.33 1.65 1.65 0 00-1 1.51V21a2 2 0 01-2 2 2 2 0 01-2-2v-.09A1.65 1.65 0 009 19.4a1.65 1.65 0 00-1.82.33l-.06.06a2 2 0 01-2.83 0 2 2 0 010-2.83l.06-.06a1.65 1.65 0 00.33-1.82 1.65 1.65 0 00-1.51-1H3a2 2 0 01-2-2 2 2 0 012-2h.09A1.65 1.65 0 004.6 9a1.65 1.65 0 00-.33-1.82l-.06-.06a2 2 0 010-2.83 2 2 0 012.83 0l.06.06a1.65 1.65 0 001.82.33H9a1.65 1.65 0 001-1.51V3a2 2 0 012-2 2 2 0 012 2v.09a1.65 1.65 0 001 1.51 1.65 1.65 0 001.82-.33l.06-.06a2 2 0 012.83 0 2 2 0 010 2.83l-.06.06a1.65 1.65 0 00-.33 1.82V9a1.65 1.65 0 001.51 1H21a2 2 0 012 2 2 2 0 01-2 2h-.09a1.65 1.65 0 00-1.51 1z"
                  />
                </svg>
              </span>
              <div
                class="jsx-887841902 text left"
              >
                Settings
              </div>
            </button>
          </div>
        </div>
        <div
          class="jsx-557291378 item xs jsx-3484918754"
        >
          <div
            class="jsx-3633143255 item jsx-478091869"
          >
            <div
              class="jsx-3563716471 item xs md jsx-3484918754"
            >
              <div
                data-testid="project-open"
              />
            </div>
          </div>
        </div>
      </div>
    </main>
  </section>
</div>
`;
